plugins/out_oci_logan/oci_logan.c

```diff
@@ -0,0 +1,1327 @@
+    },
+    {
+        FLB_CONFIG_MAP_STR, "profile_name", "DEFAULT",
+        0, FLB_TRUE, offsetof(struct flb_oci_logan, profile_name),
+        "name of the profile in the config file from which the user configs should be loaded"
+    },
+    {
+        FLB_CONFIG_MAP_BOOL, "oci_config_in_record", "false",
+        0, FLB_TRUE, offsetof(struct flb_oci_logan, oci_config_in_record),
+        "If true, oci_la_* configs will be read from the record"
+    },
+    {
+        FLB_CONFIG_MAP_STR, "uri", NULL,
+        0, FLB_TRUE, offsetof(struct flb_oci_logan, uri),
+        "Set the uri for rest api request"
+    },
+    {
+        FLB_CONFIG_MAP_STR, "oci_la_log_group_id", NULL,
+        0, FLB_TRUE, offsetof(struct flb_oci_logan, oci_la_log_group_id),
+        "log group id"
+    },
+    {
+        FLB_CONFIG_MAP_STR, "oci_la_log_set_id", NULL,
+        0, FLB_TRUE, offsetof(struct flb_oci_logan, oci_la_log_set_id),
+        ""
+    },
+    {
+        FLB_CONFIG_MAP_STR, "oci_la_entity_id", NULL,
+        0, FLB_TRUE, offsetof(struct flb_oci_logan, oci_la_entity_id),
+        ""
+    },
+    {
+        FLB_CONFIG_MAP_STR, "oci_la_entity_type", NULL,
+        0, FLB_TRUE, offsetof(struct flb_oci_logan, oci_la_entity_type),
+        ""
+    },
+    {
+        FLB_CONFIG_MAP_STR, "oci_la_log_source_name", NULL,
+        0, FLB_TRUE, offsetof(struct flb_oci_logan, oci_la_log_source_name),
+        ""
+    },
+    {
+        FLB_CONFIG_MAP_STR, "oci_la_log_set_id", NULL,
+        0, FLB_TRUE, offsetof(struct flb_oci_logan, oci_la_log_set_id),
+        ""
+    },
+    {
+        FLB_CONFIG_MAP_STR, "oci_la_log_path", NULL,
+        0, FLB_TRUE, offsetof(struct flb_oci_logan, oci_la_log_path),
+        ""
+    },
+    {
+        FLB_CONFIG_MAP_SLIST_2, "oci_la_global_metadata", NULL,
+        FLB_CONFIG_MAP_MULT, FLB_TRUE, offsetof(struct flb_oci_logan, oci_la_global_metadata),
+            ""
+    },
+    {
+        FLB_CONFIG_MAP_SLIST_2, "oci_la_metadata", NULL,
+        FLB_CONFIG_MAP_MULT, FLB_TRUE, offsetof(struct flb_oci_logan, oci_la_metadata),
+        ""
+    },
+    {
+        FLB_CONFIG_MAP_STR, "namespace", NULL,
+        0, FLB_TRUE, offsetof(struct flb_oci_logan, namespace),
+            "namespace in your tenancy where the log objects reside"
+    },
+    {
+        FLB_CONFIG_MAP_STR, "proxy", NULL,
+        0, FLB_TRUE, offsetof(struct flb_oci_logan, proxy),
+            "define proxy if required, in http://host:port format, supports only http protocol"
+    },
+
+    {0}
+};
+
+/* Plugin reference */
+struct flb_output_plugin out_oci_logan_plugin = {
+    .name           = "oci_logan",
+    .description    = "OciLogan",
+    .cb_init        = cb_oci_logan_init,
+    .cb_pre_run     = NULL,
+    .cb_flush       = cb_oci_logan_flush,
+    .cb_exit        = cb_oci_logan_exit,
+
+    /* Configuration */
+    .config_map     = config_map,
+
+    /* Events supported */
+    .event_type   = FLB_OUTPUT_LOGS | FLB_OUTPUT_TRACES,
```

Comment:
have you tested the plugin and endpoint with traces ?

Reply:
My bad, this is not supposed to support traces. Fixed.